integration_tests/build_test_containers.sh

```diff
@@ -2,11 +2,13 @@
 
 set -e
 
-echo "docker username: ${DOCKER_USERNAME}"
-docker login -u "${DOCKER_USERNAME}" -p "${DOCKER_PASSWORD}" &&
-docker pull gsyd3a/d3a:redis-staging
-
 D3A_IMAGE_TAG="d3a-tests"
+REDIS_IMAGE_NAME="gsyd3a/d3a:redis-staging"
+
+if [[ "$(docker images -q ${REDIS_IMAGE_NAME} 2> /dev/null)" == "" ]]; then
+    docker login -u "${DOCKER_USERNAME}" -p "${DOCKER_PASSWORD}" &&
+    docker pull gsyd3a/d3a:redis-staging
```

Comment:
Could use this [ENV_VARIABLE](https://github.com/gridsingularity/d3a-api-client/pull/23/files/ec29cf78968dc83470f86cc27eb2c6715dafa21c..88c208ef24206dd49c5f22aca57e08bb27191889#diff-47803b2921a6c9e72d0f98af74ac2bc5R6)